api/utils.go

```diff
@@ -329,6 +352,23 @@ func getBlocks(ctx context.Context, db *sql.DB, includeTxs bool, chainid uint64,
 			"transactionsRoot": bytesToHash(txRoot),
 			"uncles":           unclesList,
 		}
+		if intermediateBGU != nil {
```

Comment:
There are enough of these intermediate function handling examples that we should write a utility function to avoid repetition.